framework/tests/all-systems/UnionCrash.java

```diff
@@ -0,0 +1,35 @@
+// @skip-test
+public class UnionCrash {
+    void foo(MyInterface<Throwable> param) throws Throwable {
+        try {
+            bar();
+        } catch (MyExceptionA | MyExceptionB ex1) {
+            try {
+                bar();
+            } catch (SubMyExceptionA | MyExceptionB ex2) {
+// This call cause a crash because of bugs in asSuper
```

Comment:
Why the strange indentation?
Can you style this more like the usual test cases for an issue, i.e. have these lines at the beginning of the file?